### PR TITLE
Dev story/implement zustand

### PR DESCRIPTION
**Pull request by Krischan-Klug**

### CODE REVIEW

File: components/stores/moneyStore.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  persist(
+    (set) => ({
+      money: 1000,
+      addMoney: (amount) => set((state) => ({ money: state.money + amount })),
+      subtractMoney: (amount) =>
+        set((state) => ({ money: state.money - amount })),
+    }),
+    {
+      name: "money",
+      storage: createJSONStorage(() => localStorage),
```
Comment (doemser) [inline components/stores/moneyStore.js:14]:
```suggestion
```

This key is used to switch to sessionStorage, by default it is set to localStorage, so you can just remove it.

File: components/stores/petStore.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          myPets: [...state.myPets, newPet],
+        }));
+      },
+      onUpdatePet: (updatedPet) => {
+        set({
+          myPets: get().myPets.map((myPet) =>
+            myPet.id === updatedPet.id ? updatedPet : myPet
+          ),
+        });
+      },
```
Comment (doemser) [inline components/stores/petStore.js:23]:
```suggestion
        set((state) => ({
          myPets: state.myPets.map((myPet) =>
            myPet.id === updatedPet.id ? updatedPet : myPet
          ),
        }));
      },
```

We talked about it, avoid using get, use the state from inside every set function instead.

File: components/stores/petStore.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import { create } from "zustand";
+import { persist, createJSONStorage } from "zustand/middleware";
+import initialMyPets from "@/lib/initialPet.js";
+import defaultMyPet from "@/lib/myPetTemplate.js";
+
+export const usePetStore = create(
+  persist(
+    (set, get) => ({
```
Comment (doemser) [inline components/stores/petStore.js:8]:
```suggestion
    (set) => ({
```

**Review by doemser — CHANGES_REQUESTED** [review]
We already had a talk about it, you can stay with your approach, to keep it easy, but some things you should still change before going on:

- do not use get inside your stores, use the state from within the set function instead
- remove the `storage` key from the persist options object, as you are just setting the default value with it
- "stores" are react hooks and do not belong in the components folder, create a hooks folder instead

Please keep in mind that the approach you chose for the petEvents is unfavorable, this data should not be placed within a component for several reasons, you placed it there to reach the store functions, which we also already covered in our talk, so you know the pain points I have with your approach.

Please note that in a real life environment I would expect a major refactoring in order to give an approval ;)

File: components/stores/petStore.js (hunk 2 above)
Comment (Krischan-Klug) [reply components/stores/petStore.js:23]: mb, forgot this in the petStore

**Review by doemser — CHANGES_REQUESTED** [review]
Stores are still in the components folder

**Comment by Krischan-Klug** [comment]
> Stores are still in the components folder

where is the right place ? ^^

**Review by JessicaLoers — APPROVED** [review]
**Nice ✨**

**Pull request merged**